### PR TITLE
fix: extend debug options screen

### PR DESCRIPTION
**Pull request by tmspzz**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

- The presentation of the information is incorrect
- Data and functionality is missing for private builds
- Missing data and functionality is only available 

### Solutions

Extend the debug screen with the missing functionality and information

#### How to Test

On Wire Beta (release or debug) the screen is presented with the extended information.
On Wire (release) the screen is presented with minimal information

#### Before
Wire 2023-06-09 at 3_24 PM

#### After
Wire 2023-06-09 at 3_24 PM2

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: app/src/main/kotlin/com/wire/android/ui/debug/DebugDataOptions.kt [hunk 1]
```diff
@@ -17,46 +17,386 @@
...
+    val isEncryptedProteusStorageEnabled: Boolean = false,
+    val keyPackagesCount: Int = 0,
+    val mslClientId: String = "null",
+    val mlsErrorMessage: String = "null",
+    val isManualMigrationAllowed: Boolean = false
+)
+
+@Suppress("LongParameterList")
+@HiltViewModel
+class DebugDataOptionsVewModel
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/debug/DebugDataOptions.kt]:
```suggestion
class DebugDataOptionsViewModel
```
Typo 😄

File: app/src/main/kotlin/com/wire/android/ui/debug/DebugDataOptions.kt [hunk 2]
```diff
@@ -17,46 +17,386 @@
...
+                title = stringResource(R.string.debug_id),
+                text = debugId,
+                trailingIcon = R.drawable.ic_copy,
+                onIconPressed = Clickable(
+                    enabled = true,
+                    onClick = { onCopyText(debugId) }
+                )
+            )
+        }
+        if (BuildConfig.PRIVATE_BUILD) {
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/debug/DebugDataOptions.kt]: Maybe combine this `if` with the one in the line 214

File: app/src/main/kotlin/com/wire/android/ui/debug/DebugDataOptions.kt (hunk 1 above)
Comment (saleniuk) [reply app/src/main/kotlin/com/wire/android/ui/debug/DebugDataOptions.kt]: Also, we usually keep ViewModel and State data class in a separate class file

File: app/src/main/kotlin/com/wire/android/ui/debug/DebugDataOptions.kt [hunk 3]
```diff
@@ -17,46 +17,386 @@
...
+            ProteusOptions(
+                isEncryptedStorageEnabled = viewModel.state.isEncryptedProteusStorageEnabled,
+                onEncryptedStorageEnabledChange = { enabled ->
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/debug/DebugDataOptions.kt]: I would wrap it in `remember` to not recompose when not needed, or change `viewModel.enableEncryptedProteusStorage` to take `enabled` parameter like `viewModel.onEncryptedStorageEnabledChange(enabled)` and then pass it directly: `onEncryptedStorageEnabledChange = viewModel::onEncryptedStorageEnabledChange`
like it's done for other items on this screen

**Review by saleniuk — APPROVED** [review]
overall looks good, a few small suggestions but nothing huge so I approve it already 😄

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1849/1/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.0(6735113)(6735113)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.0(6735113)(6735113)-staging-compat.apk)

**Review by trOnk12 — APPROVED** [review]
agree with @saleniuk comments

File: app/src/main/kotlin/com/wire/android/ui/debug/DebugDataOptions.kt (hunk 1 above)
Comment (tmspzz) [reply app/src/main/kotlin/com/wire/android/ui/debug/DebugDataOptions.kt]: That seems like a ton of overhead and switching files for something this small :)

File: app/src/main/kotlin/com/wire/android/ui/debug/DebugDataOptions.kt (hunk 3 above)
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/debug/DebugDataOptions.kt]:
```suggestion
                onEncryptedStorageEnabledChange = remember {
                     { enabled -> 
                         if (enabled) {
                             viewModel.enableEncryptedProteusStorage()
                         }
                     }
                 }
```

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1849/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.0(6735847)(6735847)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.0(6735847)(6735847)-staging-compat.apk)

**Pull request merged**